### PR TITLE
Post link detection

### PR DESCRIPTION
**Pull request by malczak**
Adds external link detection for posts with link. Eq. See [here](https://www.facebook.com/tokfm/posts/10156119824867130)

### CODE REVIEW

File: views/page.html.pug [hunk 1]
```diff
@@ -21,3 +21,7 @@ html
           div(class="my-6") !{post.message}
           each image in post.images
             img(class="min-w-full max-w-full h-auto" src=image alt="")
+          if post.link != null
```
Comment (sobstel) [inline views/page.html.pug]: Minor: I would use `!post_link` (if it doesn't need to be strict check) or `post.link !== null` (if it needs to be strict null check).

File: builders/buildPost.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Post } from "../models";
+import { Post, PostLink } from "../models";
```
Comment (sobstel) [inline builders/buildPost.ts:1]: What do you think about naming it `SourceLink` or `FacebookLink` or similar to more clearly indicate that it's facebook link (and not calmbook interrnal post link).

File: builders/buildPost.ts [hunk 3]
```diff
@@ -21,14 +21,16 @@ const buildPost = ($: CheerioSelector): Post => {
       .slice(0, 8)
       .join(" ") + "...";
 
-  return { title, id, timestamp, message, images };
+  console.log(link);
```
Comment (sobstel) [inline builders/buildPost.ts]: `console.log` to remove again ;-) maybe we should add some pre-build step to check those at some point ;-)

File: builders/buildPost.ts [hunk 4]
```diff
@@ -51,5 +53,27 @@ const buildContent = (
   const images =
     imgNodes && imgNodes.toArray().map(element => element.attribs["src"]);
 
-  return { message, images };
+  // fetch any external links
+  let postLink: PostLink | undefined = undefined;
```
Comment (sobstel) [inline builders/buildPost.ts:55]: This method is doing a bit too much already. What do you think about extracting `buildMessage`, `buildMessage`, `buildPostLink` methods? They could accept `userContent` as an only argument.

**Review by sobstel — CHANGES_REQUESTED** [review]

File: builders/buildPost.ts (hunk 4 above)
Comment (malczak) [reply builders/buildPost.ts:55]: I think we can make this a separate task, not only to split that method but also to refactor post scrapper (eq. by adding post types - text, video, link, image etc)

File: builders/buildPost.ts (hunk 2 above)
Comment (malczak) [reply builders/buildPost.ts:1]: Lets put this as part of post scrapper creation task

File: builders/buildPost.ts (hunk 4 above)
Comment (sobstel) [reply builders/buildPost.ts:55]: 👍  ok, just please create a ticket for this before merging this PR. thanks in advance.

Comment (malczak) [reply builders/buildPost.ts:55]: Needs to be discussed  #13

**Review by sobstel — APPROVED** [review]

**Pull request merged**